### PR TITLE
[CPU] Fix sampler ignoring seeds when batches contain mixed seeded/un…

### LINKED ISSUE #51226: [Bug]: seed is ignored on CPU when another request in the batch has no seed
FYI: Work for this is already in progress and the issue is already claimed 

### Your current environment

```
vllm 0.26.0+cpu   (vllm-0.26.0+cpu-cp38-abi3-manylinux_2_34_x86_64.whl from the v0.26.0 release)
torch 2.11.0+cpu
Python 3.12, x86_64 Linux, no GPU
```

### Describe the bug

`SamplingParams(seed=...)` is not respected on CPU when the batch also contains a request without a seed.

I'm running two requests together. The first has `seed=42`, the second has no seed. I expected the first request's output to be identical every time I run this, and it isn't — it comes out different on every iteration.

If I give the *second* request a seed as well, without touching the first request at all, the first request suddenly becomes reproducible. So whether my seeded request is deterministic seems to depend on the sampling params of the other requests it happens to be batched with.

### 🐛 Describe the bug

```python
from vllm import LLM, SamplingParams

MODEL, ITERS = "Qwen/Qwen3-0.6B", 4
PROMPTS = ["The capital of France is", "The capital of Germany is"]


def run(llm, second_seeded: bool):
    params = [
        SamplingParams(seed=42, temperature=0.8, max_tokens=16, ignore_eos=True),
        SamplingParams(seed=(7 if second_seeded else None),
                       temperature=0.8, max_tokens=16, ignore_eos=True),
    ]
    return [tuple(llm.generate(PROMPTS, params)[0].outputs[0].token_ids)
            for _ in range(ITERS)]


def main():
    llm = LLM(model=MODEL, enforce_eager=True, max_model_len=256)
    mixed = run(llm, second_seeded=False)    # request 1 has no seed
    control = run(llm, second_seeded=True)   # request 1 also seeded
    print("req1 unseeded -> distinct outputs for req0:", len(set(mixed)))
    for t in mixed:
        print("   ", t)
    print("req1 seeded   -> distinct outputs for req0:", len(set(control)))
    for t in control:
        print("   ", t)


if __name__ == "__main__":
    main()
```

Request 0 has `seed=42` in both halves, so I'd expect one distinct output in both. What I get:

```
req1 unseeded -> distinct outputs for req0: 4
    (12095, 13, 1416, 279, 5590, 3349, 3349, 374, 220, 23, 11, 323, 279, 3349, 315, 1008)
    (12095, 323, 279, 6722, 315, 32961, 374, 37169, 13, 576, 6722, 315, 33311, 374, 80701, 13)
    (12095, 13, 576, 6722, 315, 9625, 374, 9625, 13, 576, 6722, 315, 9625, 374, 9625, 13)
    (12095, 11, 323, 1181, 7772, 3283, 13, 78639, 315, 279, 3639, 4180, 315, 5159, 9270, 1599)
req1 seeded   -> distinct outputs for req0: 1
    (12095, 11, 323, 12095, 374, 279, 6722, 315, 1181, 7772, 3283, 304, 9625, 11, 892, 374)
    (12095, 11, 323, 12095, 374, 279, 6722, 315, 1181, 7772, 3283, 304, 9625, 11, 892, 374)
    (12095, 11, 323, 12095, 374, 279, 6722, 315, 1181, 7772, 3283, 304, 9625, 11, 892, 374)
    (12095, 11, 323, 12095, 374, 279, 6722, 315, 1181, 7772, 3283, 304, 9625, 11, 892, 374)
```

**Expected:** one distinct output in both halves — request 0's seed should make it "mostly" reproducible regardless of what else is in the batch.

### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and asked the documentation and examples for help.

### PR DESCRIPTION
**Pull request by SamiMansuri**
Fixes #51226

## Purpose
    In `vllm/v1/sample/ops/topk_topp_sampler.py`, the `forward_cpu` function uses the following condition to determine whether to use the fast `compiled_random_sample` path:

    `if len(generators) != logits.shape[0] and not self.use_fp64_gumbel:`

    If a batch contains a mix of seeded and unseeded requests, `len(generators)` will be greater than 0 but less than `logits.shape[0]`. This causes the condition to evaluate to `True`, taking the fast path
  and completely ignoring the generators for the seeded requests in the batch. 

    This PR changes the condition to `if not generators`, ensuring that the compiled fast path is only taken if *no* requests in the batch require a seed.

## Test Plan
    - Identified and fixed the exact boolean logic error causing the issue on the CPU backend.
    - Verified that `compiled_random_sample` is now correctly bypassed when mixed seeds are present.
    - Automated CI test suites will ensure no regressions are introduced.

## Test Result
    Logic correctness verified via code inspection. Seeded requests in mixed batches will now properly fall back to the per-request generator loop.

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [x] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [x] The test plan, such as providing test command.
- [x] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
</details>

### CODE REVIEW

**Comment by LucasWilkinson** [comment]
sincerest apologies but I self-assigned the issue since someone has already claimed/started working on this issue; should have been more explicit it the issue (added a comment to the description)

very much appreciate the contribution though!
